devices/ICub/src/ICub.cpp

```diff
@@ -0,0 +1,388 @@
+
+         pImpl->ftSensorsMap.emplace(
+                     ft6dPrefix + pImpl->sensorNames[s],
+                     ICubImpl::icubDeviceSensors<ICubImpl::ICubForceTorque6DSensor>{ft6d,s});
+    }
+
+    return true;
+}
+
+// ---------------
+// Generic Methods
+// ---------------
+
+wearable::WearableName ICub::getWearableName() const
+{
+    return pImpl->wearableName;
+}
+
+wearable::WearStatus ICub::getStatus() const
+{
+    wearable::WearStatus status = wearable::WearStatus::Ok;
+
+    for (const auto& s : getAllSensors()) {
+        if (s->getSensorStatus() != sensor::SensorStatus::Ok) {
+            yError() << LogPrefix << "The status of" << s->getSensorName() << "is not Ok ("
+                     << static_cast<int>(s->getSensorStatus()) << ")";
+            status = wearable::WearStatus::Error;
+        }
+    }
+
+    // Default return status is Ok
+    return status;
+}
+
+
+wearable::TimeStamp ICub::getTimeStamp() const
+{
+    // Stamp count should be always zero
+    return {pImpl->timeStamp.time,0};
+}
+
+bool ICub::close()
+{
+    return true;
+}
+
+// =========================
+// IPreciselyTimed interface
+// =========================
+yarp::os::Stamp ICub::getLastInputStamp()
+{
+    // Stamp count should be always zero
+    return yarp::os::Stamp(0, pImpl->timeStamp.time);
+}
+
+// ---------------------------
+// Implemented Sensors Methods
+// ---------------------------
+
+wearable::SensorPtr<const wearable::sensor::ISensor>
+ICub::getSensor(const wearable::sensor::SensorName name) const
+{
+    wearable::VectorOfSensorPtr<const wearable::sensor::ISensor> sensors = getAllSensors();
+    for (const auto& s : sensors) {
+        if (s->getSensorName() == name) {
+            return s;
+        }
+    }
+    yWarning() << LogPrefix << "User specified name <" << name << "> not found";
+    return nullptr;
+}
+
+wearable::VectorOfSensorPtr<const wearable::sensor::ISensor>
+ICub::getSensors(const wearable::sensor::SensorType type) const
+{
+    wearable::VectorOfSensorPtr<const wearable::sensor::ISensor> outVec;
+    switch (type) {
+        case sensor::SensorType::ForceTorque6DSensor: {
+            outVec.reserve(pImpl->nSensors);
+            for (const auto& ft6d : pImpl->ftSensorsMap) {
+                outVec.push_back(
+                    static_cast<std::shared_ptr<sensor::ISensor>>(ft6d.second.icubSensor));
+            }
+
+            break;
+        }
+        default: {
+            //yWarning() << LogPrefix << "Selected sensor type (" << static_cast<int>(type)
+            //           << ") is not supported by ICub";
+            return {};
+        }
+    }
+
+    return outVec;
+}
+
+wearable::SensorPtr<const wearable::sensor::IForceTorque6DSensor>
+ICub::getForceTorque6DSensor(const wearable::sensor::SensorName name) const
+{
+    // Check if user-provided name corresponds to an available sensor
+    if (pImpl->ftSensorsMap.find(static_cast<std::string>(name))
```

Comment:
Are you sure that the static cast is required? `SensorName` is just a typedef of string.

Reply:
You are correct, this casting is not necessary